### ISSUE TITLE
Inclusion of common glyphs in text

### ISSUE THREAD

**Issue by joeberkovitz** [w3c/mnx#42]
How should the inclusion of glyphs like [flat] and [sharp] in runs of text be handled in CWMNX? Part name display is a case where this is needed.

In MusicXML one must switch between <display-text> and <accidental-text> to make this happen in  some circumstances, or switch between fonts. With SMuFL in the picture there should be an easier way to approach this.

**Comment by mdgood** [comment]
MusicXML 3.1 adds the `<symbol>` and `<credit-symbol>` elements which can be interspersed with `<words>` and `<credit-words>` elements respectively. The content of these elements are SMuFL glyph names. This seems like a good starting point - more modern and comprehensive than the `<display-text>` and `<accidental-text>` approach - though MNX will likely remove the distinction between `<symbol>` and `<credit-symbol>`. It's not just an issue for common glyphs, it's an issue for any musical symbol.

**Comment by joeberkovitz** [comment]
[Yes, I intended this to be an issue for every musical symbol, not just accidentals.]

What are people's thoughts on the direct inclusion of SMuFL code points in the document as Unicode characters, to indicate "code shifting" to musical glyphs in a string? In other words, to represent _B [flat]_, encode this with a numeric entity escape as `B&#E260;`

This would be very handy for MNX as it would work in element attributes as well as in text content.

**Comment by mdgood** [comment]
That alas won't work for SMuFL stylistic alternates which have no standard code point, but do have a canonical glyph name. It is also more opaque to the human reader, especially since SMuFL uses the Unicode private area rather than being part of the Unicode standard.

**Comment by siennamw** [comment]
Yes, it is definitely

> more opaque to the human reader

and also seems like a step away from semantic encoding.  The code point could represent any symbol - a clef, an articulation symbol, a rest - and being able to use these symbols where they have no semantic meaning (ex. a rest where a clef should be) would destroy the semantic value of a specification for CWMN.

That said, this type of freedom might be valuable in GMNX because changing or expanding the semantic meanings of symbols often occurs in experimental music notations.

**Comment by joeberkovitz** [comment]
@siennamw In general I agree with the principle you're stating, but this specific case is about including glyphs in runs of text like part names or titles, which are not supposed to be treated semantically any more than, say, lyrics.

@mdgood Overall I'm trying to see if we can get to a place where _simple text_ in MNX (e.g "Bb Clarinet" for a partname, where the `b` is an actual flat glyph) can be expressed with as little complexity as possible. I feel that having to reference the symbolic names of SMuFL stylistic alternates in XML elements inside a run of text is a bit grungy and low-level. But maybe it's the best we can do... let's see what else gets proposed.

**Comment by siennamw** [comment]
Yes, that makes more sense, and I do see that it says 'text' in the issue title.

I don't agree that

> having to reference the symbolic names of SMuFL stylistic alternates in XML elements inside a run of text is a bit grungy and low-level

I think something like `B<smufl glyph='accidentalFlat' /> Clarinet` is much clearer and more human-readable than a code point with no other context.  I suppose we could also allow `<smufl codepoint='U+E260' />`, but this is significantly less human-readable than the previous example.

**Comment by hhpmusic** [comment]
Yes, using the direct glyph names may make both the human readers and braille music transcription softwares much more easier to interpret.

**Comment by clnoel** [comment]
Even though it hurts human-readability, I lean toward using the Unicode form<smufl codepoint='U+E260' />. A machine can interpret the Unicode symbol much more directly than the text form, which will require a conversion table.

Alternatively, if we want to improve human-readability while preserving machine-readability, we can specify a new "glyph" tag, and put define elsewhere that certain glyphs are associated with certain smufl codes. For example:

```
<glyph-definition id="flat" smufl-codepoint="U+E260"/>

...

B<glyph id="flat"> Clarinet
```
